sopel/config/types.py

```diff
@@ -194,7 +195,9 @@ def configure(self, prompt, default, parent, section_name):
             raise ValueError("You must provide a value for this option.")
 
         value = value or default
-        return self.parse(value)
+
+        section = getattr(parent, section_name)
+        return self._parse(value, parent, section)
```

Comment:
Seems weird to have assignment, blank line, assignment, use both assignments in return. To me it makes more sense to have assignment, assignment, blank line, use assigned values.

```suggestion
        section = getattr(parent, section_name)

        return self._parse(value, parent, section)
```

(Nitpick mode: Active. 😝)

Reply:
I think I was too tired when I wrote all of these, it's not the first comment where I think I should have known better.